Mage/src/main/java/mage/watchers/common/AttackingBlockingWatcher.java

```diff
@@ -0,0 +1,68 @@
+package mage.watchers.common;
+
+import mage.constants.WatcherScope;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.watchers.Watcher;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * Stores attacking/blocking combat information but is only updated
+ *  1) as damage is dealt (combat damage or otherwise)
+ *  2) as a spell or ability starts resolving
+ *  3) Land playing or special action taken (just in case that then causes a static effect to kill)
+ * Thus the information is available after any involved creatures die
+ * and all dying creatures can see all other creatures that were in combat at that time
+ * @author notgreat
+ */
+public class AttackingBlockingWatcher extends Watcher {
+
+    private Set<UUID> attackers = new HashSet<>();
+    private Set<UUID> blockers = new HashSet<>();
+
+    public AttackingBlockingWatcher() {
+        super(WatcherScope.GAME);
+    }
+    @Override
+    public void watch(GameEvent event, Game game) {
+        switch (event.getType()) {
+            case LAND_PLAYED:
+            case TAKEN_SPECIAL_ACTION:
+            case RESOLVING_ABILITY:
```

Comment:
just confirming this event does get fired for spells?

Reply:
I can confirm that `RESOLVING_ABILITY` includes spells, `SpellAbility` inherits it from `AbilityImpl` in the `resolve` function.